Airwallex/Core/Sources/Internal/AWXNetWorkManager.swift

```diff
@@ -0,0 +1,187 @@
+        method: HTTPMethod = .GET,
+        parameters: [String: Any]? = nil,
+        payload: Data? = nil,
+        eventName: String,
+        completion: @escaping (Result<T, Error>) -> Void
+    ) {
+        var request: URLRequest
+        if method == .GET {
+            var urlComponents = URLComponents(string: baseURL + path)
+            var queryItems = [URLQueryItem]()
+            if let parameters = parameters as? [String: String] {
+                for (key, value) in parameters {
+                    queryItems.append(URLQueryItem(name: key, value: value))
+                }
+            }
+            urlComponents?.queryItems = queryItems
+            guard let url = urlComponents?.url else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Invalid URL format.", comment: "Invalid URL format."))))
+                }
+                return
+            }
+            request = URLRequest(url: url)
+        } else if method == .POST {
+            guard let url = URL(string: baseURL + path) else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Invalid URL format.", comment: "Invalid URL format."))))
+                }
+                return
+            }
+            request = URLRequest(url: url)
+            request.httpBody = payload
+        } else {
+            guard let url = URL(string: baseURL + path) else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Invalid URL format.", comment: "Invalid URL format."))))
+                }
+                return
+            }
+            request = URLRequest(url: url)
+        }
+
+        request.httpMethod = method.rawValue
+
+        for (key, value) in headers {
+            request.setValue(value, forHTTPHeaderField: key)
+        }
+
+        if let clientSecret = AWXAPIClientConfiguration.shared().clientSecret {
+            request.setValue(clientSecret, forHTTPHeaderField: "client-secret")
+        } else {
+            logMessage("Client secret is not set!")
+        }
+
+        logMessage("ULR request: \(request.url?.absoluteString ?? "")")
+        let task = session.dataTask(with: request) { [weak self] data, response, error in
+            guard let url = request.url else {
+                let message = NSLocalizedString("No URL.", comment: "No URL.")
+                self?.logMessage(message)
+                DispatchQueue.main.async {
+                    completion(.failure(NSError.errorForAirwallexSDK(with: message)))
+                }
+                return
+            }
+
+            if let error {
```

Comment:
Will this error be contradicted/overlapped with the `if let error = T.parseError(data)` check down below? Because when error happens during the urlsession, it could have error response `AWXAPIErrorResponse` at the same time.